### PR TITLE
*: Update TiDB version, check ClientLocalFiles capability when handling load data operation

### PR DESCRIPTION
**Pull request by zimulala**
see: [CLIENT_LOCAL_FILES](https://dev.mysql.com/doc/internals/en/capability-flags.html#flag-CLIENT_LOCAL_FILES)

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @iamxy @shenli

File: server/packetio.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func (p *packetIO) readOnePacket() ([]byte, error) {
 
 	sequence := uint8(header[3])
 	if sequence != p.sequence {
-		return nil, errInvalidSequence.Gen("invalid sequence %d != %d", sequence, p.sequence)
+		return nil, errNotAllowedCommand.Gen("invalid sequence %d != %d", sequence, p.sequence)
```
Comment (shenli) [inline server/packetio.go]: Could you add a test case for this error?

**Comment by zimulala** [comment]
PTAL @shenli @coocood

**Comment by coocood** [comment]
LGTM

File: server/conn.go [hunk 2]
```diff
@@ -53,6 +53,9 @@ import (
 	"github.com/pingcap/tidb/util/hack"
 )
 
+// mockLoadDataClientLocalFilesCap uses to mock ClientLocalFiles capability.
+var mockLoadDataClientLocalFilesCap bool
+
 var defaultCapability = mysql.ClientLongPassword | mysql.ClientLongFlag |
```
Comment (tiancaiamao) [inline server/conn.go]: add ClientLocalFiles to defaultCapability

File: server/server_test.go [hunk 3]
```diff
@@ -207,6 +207,7 @@ func runTestPreparedString(t *C) {
 func runTestLoadData(c *C) {
 	path := "/tmp/load_data_test.csv"
 	runTests(c, dsn+"&allowAllFiles=true", func(dbt *DBTest) {
```
Comment (tiancaiamao) [inline server/server_test.go]: try to use "&allowAllFiles=true" to control client's capability and avoid mockLoadDataClientLocalFilesCap

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**